lms/djangoapps/instructor/enrollment.py

```diff
@@ -229,16 +231,25 @@ def reset_student_attempts(course_id, student, module_state_key, delete_module=F
                 except StudentModule.DoesNotExist:
                     # If a particular child doesn't have any state, no big deal, as long as the parent does.
                     pass
+        # Some blocks (openassessment) use StudentModule data as a key for internal submission data.
+        # Inform these blocks of the reset and allow them to handle their data.
+        clear_submission = getattr(block, "clear_submission", None)
+        if callable(clear_submission):
+            clear_submission(
+                user_id=user_id,
+                course_id=course_id.to_deprecated_string(),
+                item_id=module_state_key.to_deprecated_string()
+            )
+            submission_cleared = True
     except ItemNotFoundError:
         log.warning("Could not find %s in modulestore when attempting to reset attempts.", module_state_key)
 
-    # Reset the student's score in the submissions API
-    # Currently this is used only by open assessment (ORA 2)
+    # Reset the student's score in the submissions API, if the xblock.clear_submission() above has not done so already
     # We need to do this *before* retrieving the `StudentModule` model,
     # because it's possible for a score to exist even if no student module exists.
-    if delete_module:
+    if delete_module and not submission_cleared:
```

Comment:
Agreed. Our other options here (none of which I'm particularly fond of) are:
* don't have the `clear_submission` implementation defined by `openassessment` call `reset_score()`, and let this line handle it. That seems to be a step in reverse, by having the LMS be aware of *more* things than before.
* let `reset_score()` be called twice, once by `clear_submission` and once here. Now that I look deeper into how this would work, it seems like this would be handled pretty cleanly from the submissions side of things: https://github.com/edx/edx-submissions/blob/master/submissions/api.py#L646. If anyone is strongly in favor of this double-call instead of the complexity of tracking `submission_cleared`, I could be convinced.
* make sure all xblocks that use submissions move to the `clear_submission` passthrough paradigm implemented by `openassessment`. This would be ideal, and I thought it was what we were doing here, but with other xblocks using submissions I guess a gradual deprecation is the best we can hope for.